README.md

```diff
@@ -2,4 +2,88 @@
+
+The package is currently configured to deploy to [the public `hadoop` Artifacts feed](https://dev.azure.com/endjin-labs/hadoop/_artifacts/feed/hadoop) in the `endjin-labs` Azure DevOps instance as `com.endjin.hadoop:hadoop-azure-token-providers`.
+
+You need to provide credentials in a settings file to deploy to the feed. Create a `settings.xml` file in the `~/.m2` folder with the following contents:
+
+```xml
+<settings>
+  <servers>
+    <server>
+      <id>hadoop</id>
+      <username>endjin-labs</username>
+      <password>{Personal Access Token}</password>
+    </server>
+  </servers>
+</settings>
+```
+
+Replace `{Personal Access Token}` with a Personal Access Token generated from the `endjin-labs` Azure DevOps instance with permission to read/write Packages.
+
+In the `pom.xml` file, set the desired version number in the `<version>` tag, then 
+use Maven to deploy the package:
+
+`mvn deploy`
+
+> TODO: Add GitHub Actions workflow for automatically versioning & deploying the package
+
+### Usage from PySpark
+
+The package was built in order to enable a simple way of accessing data in Azure Data Lake when developing a PySpark solution locally in a Dev Container.
+
+To use the custom token providers, include the `org.apache.hadoop:hadoop-azure` and `com.endjin.hadoop:hadoop-azure-token-providers` packages when creating your Spark session. You will also need to add [the `endjin-labs` `hadoop` Artifacts feed](https://dev.azure.com/endjin-labs/hadoop/_artifacts/feed/hadoop) as an extra Maven repository.
+
+e.g.
+
+```python
+builder = (
+    SparkSession.builder.appName('my-app')
+    .master("local[*]")
+    .config(
+        "spark.jars.repositories", 
+        "https://pkgs.dev.azure.com/endjin-labs/hadoop/_packaging/hadoop/maven/v1"
+    )
+)
+
+extra_packages = [
+    'org.apache.hadoop:hadoop-azure:3.3.3',
+    'com.endjin.hadoop:hadoop-azure-token-providers:1.0.1'
+]
+
+spark = configure_spark_with_delta_pip(builder, extra_packages=extra_packages).getOrCreate()
+```
+
+To configure one of the custom token providers, set the relevant [Hadoop configuration](https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html#Custom_OAuth_2.0_Token_Provider), e.g. for `AzureCliCredentialTokenProvider`:
+
+```python
+spark.sparkContext._jsc.hadoopConfiguration().set(
+    "fs.azure.account.auth.type",
+    "Custom"
+)
+spark.sparkContext._jsc.hadoopConfiguration().set(
+    "fs.azure.account.oauth.provider.type",
+    "com.endjin.hadoop.fs.azurebfs.custom.AzureCliCredentialTokenProvider"
+)
+```
+
+This will then use the token provider by default for getting access tokens for all Azure Storage accounts.
+
+#### `AzureCliCredentialTokenProvider`
+
+This token provider relies on the credentials from the Azure CLI in order to retrieve an access token for Azure Storage. When configured, ensure that `az login` has been run (for the relevant tenant) before attempting to use Spark to access data in the Data Lake / Storage Account.
```

Comment:
How about a winget instruction for installing the Azure CLI?

Reply:
I've added a link to installation instruction page